src/sage/combinat/symmetric_group_representations.py

```diff
@@ -1015,3 +1016,568 @@ def partition_to_vector_of_contents(partition, reverse=False):
+            return self.subrepresentation([], is_closed=True)
+        return self._graded_decomposition[k]
+
+    def graded_representation_matrix(self, elt, q=None):
+        r"""
+        Return the matrix corresponding to the left action of the symmetric
+        group (algebra) element ``elt`` on ``self``.
+
+        EXAMPLES::
+
+            sage: SGA = SymmetricGroupAlgebra(GF(3), 3)
+            sage: GP = SGA.garsia_procesi_module([1, 1, 1])
+            sage: elt = SGA.an_element(); elt
+            [1, 2, 3] + 2*[1, 3, 2] + [3, 1, 2]
+            sage: X = GP.graded_representation_matrix(elt); X
+            [  0   0   0   0   0   0]
+            [  0 q^2   0   0   0   0]
+            [  0 q^2   0   0   0   0]
+            [  0   0   0   q   0   0]
+            [  0   0   0   q   0   0]
+            [  0   0   0   0   0   1]
+            sage: X.parent()
+            Full MatrixSpace of 6 by 6 dense matrices over
+             Univariate Polynomial Ring in q over Finite Field of size 3
+            sage: R.<q> = GF(3)[]
+            sage: t = R.quotient([q^2+2*q+1]).gen()
+            sage: GP.graded_representation_matrix(elt, t)
+            [       0        0        0        0        0        0]
+            [       0 qbar + 2        0        0        0        0]
+            [       0 qbar + 2        0        0        0        0]
+            [       0        0        0     qbar        0        0]
+            [       0        0        0     qbar        0        0]
+            [       0        0        0        0        0        1]
+        """
+        if q is None:
+            q = self.base_ring()['q'].gen()
+        R = q.parent()
+        return matrix(R, [q**b.degree() * (elt * b).to_vector().change_ring(R)
+                          for b in self.basis()])
+
+    def graded_brauer_character(self):
+        r"""
+        Return the graded Brauer character of ``self``.
+
+        EXAMPLES::
+
+            sage: SGA = SymmetricGroupAlgebra(GF(2), 5)
+            sage: GP311 = SGA.garsia_procesi_module([3, 1, 1])
+            sage: GP311.graded_brauer_character()
+            (6*q^3 + 9*q^2 + 4*q + 1, q + 1, q^3 - q^2 - q + 1)
+        """
+        q = QQ['q'].gen()
+        return sum(q**d * SM.brauer_character() for d, SM in self._graded_decomposition.items())
+
+    class Element(QuotientRing_generic.Element):
+        def _acted_upon_(self, scalar, self_on_left=True):
+            r"""
+            Return the action of ``scalar`` on ``self``.
+
+            EXAMPLES::
+
+                sage: SGA = SymmetricGroupAlgebra(GF(3), 4)
+                sage: GP22 = SGA.garsia_procesi_module([2, 2])
+                sage: x = SGA.an_element(); x
+                [1, 2, 3, 4] + 2*[1, 2, 4, 3] + [4, 1, 2, 3]
+                sage: v = GP22.an_element(); v
+                -gp1 - gp2 - gp3
+                sage: g = SGA.group().an_element(); g
+                [4, 1, 2, 3]
+                sage: g * v
```

Comment:
```suggestion
                sage: g * v  # indirect doctest
```

Reply:
While not necessary for these tests (since it is a `_sage_` method), I've added it here and on similar tests.